### DISCUSSION TITLE
Let's talk about styling

### DISCUSSION THREAD

**Discussion by M1ck0** [vercel/next.js#23464] [Help]
I just started using `styled-components` and it is really great, mostly because you can pass props to CSS and render what you need when you need it. No more messy `className` things.

It also works on the server-side which is great.

But I have a problem with few things.

### 1. How to load fonts on the server-side?
Currently I am loading `styled-components` using [this](https://github.com/vercel/next.js/tree/master/examples/with-styled-components). 
Problem is that I don't know how to load fonts on the server. There is a brief flash of all text on the page before a font is loaded. This is really noticeable on slower connections (such as mine :D)

~~### 2.Inputs are acting strange
For some reason when the page is loaded for the first time everything is alright with inputs. But after pressing `F5` or `Ctrl+R` all inputs briefly flash. Like they have `background-color` set to `#000` but there is no way I can see what is going on.~~

### 3. How to load `.css` files on the server-side?
I have some global styles (normalize.css and grid system that I use) and these are not loaded on the server side. This is a problem because whole content jumps before these styles are loaded.
Currently, I am login them in `_app.js` using `import "../normalize.css"`

My `.babelrc` configuration is pretty simple.
```json
{
  "presets": ["next/babel"],
  "plugins": [
    [
      "babel-plugin-styled-components",
      {
        "ssr": true,
        "pure": true
      }
    ]
  ]
}
```

<hr/>

Are these two problems even possible to fix? 
Is `styled-components` approach wrong?

What is the best way in your opinion to write `CSS/SCSS` for `Next` applications?

Thank you for reading.

**Comment by jaisharx** [comment]
Hey, it depends on how you're currently loading fonts, but if you are using @font-face css rule then you can just try putting it in the createGlobalStyle sheet tag styled-components provides (the _app.js example has it). You can also place your normalize reset inside it (you can even create a normalize.js file exporting a globalStyleSheet and then can simply use it in _app.js). I think the reason for the layout jump is that your normalize css is external while all the css generated by styled-components is inlined. Good luck, hope it fixes your issues.
